app/models/report.rb

```diff
@@ -14,4 +36,17 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  def create_mention_list
+    mention_from_me.delete_all
+    mentioned_ids = content.scan(%r{reports/(\d+)}).flatten.map(&:to_i)
+    return if mentioned_ids.empty?
+
+    mentioned_ids.uniq.each do |target_id|
+      next if target_id == id
+      next unless Report.exists?(target_id)
+
+      ReportMention.create!(source_report_id: id, target_report_id: target_id)
```

Comment:
```suggestion
      mentioning_report = Report.where.not(id: id).find_by(id: target_id)
      if mentioning_report
        mentioning_reports << mentioning_report
      end
```

こんな感じでも同じことが実現できるかもです

Reply:
ご提案ありがとうございます。
こちら[55493d2](https://github.com/koguchi-e/fjord-books_app-2023/pull/6/commits/55493d2d0c0906e52e4ea13e87f87c5ff933abcb)にて修正しました。

ご提案のコードですと、`mentioning_reports` の関連も同時に更新されるので、
メモリ上の関連キャッシュが即時反映されるということがわかりました。